test/util/framework/per_test_framework.go

```diff
@@ -236,16 +262,41 @@ func (tc *perItOrDescribeTestContext) NewResourceGroup(ctx context.Context, reso
 // cleanupResourceGroup is the standard resourcegroup cleanup.  It attempts to
 // 1. delete all HCP clusters and wait for success
 // 2. delete the resource group and wait for success
-func cleanupResourceGroup(ctx context.Context, hcpClient *hcpsdk20240610preview.HcpOpenShiftClustersClient, resourceGroupsClient *armresources.ResourceGroupsClient, resourceGroupName string) error {
+func cleanupResourceGroup(ctx context.Context, hcpClient *hcpsdk20240610preview.HcpOpenShiftClustersClient, resourceGroupsClient *armresources.ResourceGroupsClient, resourceGroupName string, timeout time.Duration) error {
 	errs := []error{}
 
 	ginkgo.GinkgoLogr.Info("deleting all hcp clusters in resource group", "resourceGroup", resourceGroupName)
-	if err := DeleteAllHCPClusters(ctx, hcpClient, resourceGroupName, 60*time.Minute); err != nil {
+	if err := DeleteAllHCPClusters(ctx, hcpClient, resourceGroupName, timeout); err != nil {
 		return fmt.Errorf("failed to cleanup resource group: %w", err)
 	}
 
 	ginkgo.GinkgoLogr.Info("deleting resource group", "resourceGroup", resourceGroupName)
-	if err := DeleteResourceGroup(ctx, resourceGroupsClient, resourceGroupName, 60*time.Minute); err != nil {
+	if err := DeleteResourceGroup(ctx, resourceGroupsClient, resourceGroupName, false, timeout); err != nil {
+		return fmt.Errorf("failed to cleanup resource group: %w", err)
+	}
+
+	return errors.Join(errs...)
+}
+
+// cleanupResourceGroupNoRP performs cleanup when the resource provider is not available
+// This is used to cleanup personal dev e2e test runs, where the infra is already gone so there's no
+// RP to call for HCP deletion.
+// 1. deletes the "*--managed" resource group. Use 'force' flag to speed up VM deletion.
```

Comment:
Is the cleanup only intended to be used against personal dev env? If it's intended for Stage/Prod just be aware that there is going to be a deny assignment that'll prevent the deletion of the MRG.